diff -ruNa origin/widget/windows/nsWindow.cpp now/widget/windows/nsWindow.cpp
--- origin/widget/windows/nsWindow.cpp	Wed Mar 20 21:26:52 2013
+++ now/widget/windows/nsWindow.cpp	Sat Mar 23 20:21:33 2013
@@ -617,7 +617,7 @@
     // bugs over the years, disable it (sTrimOnMinimize=1) on Vista and up.
     sTrimOnMinimize =
       Preferences::GetBool("config.trim_on_minimize",
-        (WinUtils::GetWindowsVersion() >= WinUtils::VISTA_VERSION)) ? 1 : 0;
+        false);
     sSwitchKeyboardLayout =
       Preferences::GetBool("intl.keyboard.per_window_layout", false);
   }
@@ -1630,6 +1630,11 @@
         // If sTrimOnMinimize is set, we let windows handle minimization normally
         // using SW_MINIMIZE.
         mode = sTrimOnMinimize ? SW_MINIMIZE : SW_SHOWMINIMIZED;
+		if (sTrimOnMinimize) {
+			HANDLE hProcess = GetCurrentProcess();
+			if(hProcess != NULL)
+				SetProcessWorkingSetSize(hProcess, -1, -1);
+		}
         break;
 
       default :
